support-models/src/main/scala/com/gu/support/catalog/ZuoraCatalog.scala

```diff
@@ -0,0 +1,31 @@
+package com.gu.support.catalog
+
+import io.circe._
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+
+case class ZuoraCatalog(
+    products: Array[ZuoraProduct],
```

Comment:
Use List rather than Array for all these iterable types. Arrays in Scala are mutable which we don't want

Reply:
Ah, thanks – I hadn’t realised Arrays are mutable. We chose it over List because I was thinking that linked lists would lead to poor cache locality for large collections like this. For this reason, I’m now leaning towards Vector, but since I know Scala types less well I reckon I should ask: do you think that’s worthwhile?

Reply:
I would still say use List since all we are doing is mapping over it. Whilst Vector is more performant for quite a few operation types, realistically performance in an application of this kind will not be affected by a choice of a collection implementation so keeping consistency with the rest of the codebase (which uses List almost exclusively) is likely of more value.

Reply:
Ok, that sounds good

Reply:
Great, ported to List now and will do going forwards.